scipy/optimize/_linprog.py

```diff
@@ -506,21 +506,18 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
 
     # Keep the original arrays to calculate slack/residuals for original
     # problem.
-    c_o, A_ub_o, b_ub_o, A_eq_o, b_eq_o = c.copy(
-    ), A_ub.copy(), b_ub.copy(), A_eq.copy(), b_eq.copy()
+    lp_o = lp
 
     # Solve trivial problem, eliminate variables, tighten bounds, etc...
     c0 = 0  # we might get a constant term in the objective
     if solver_options.pop('presolve', True):
         rr = solver_options.pop('rr', True)
-        (c, c0, A_ub, b_ub, A_eq, b_eq, bounds, x, x0, undo, complete, status,
-            message) = _presolve(c, A_ub, b_ub, A_eq, b_eq, bounds, x0, rr, tol)
+        (lp, c0, x, undo, complete, status, message) = _presolve(lp, rr, tol)
 
-    postsolve_args = (c_o, A_ub_o, b_ub_o, A_eq_o, b_eq_o, bounds, undo)
+    postsolve_args = (lp_o.c, lp_o.A_ub, lp_o.b_ub, lp_o.A_eq, lp_o.b_eq, lp.bounds, undo)
```

Comment:
Maybe put the whole `lp_o` in and unpack it later?

Reply:
I was thinking about this as well for consistency, but I wasn't sure if it's a good idea to make `postsolve_args` nested, ie. 
```Python
# definition:
postsolve_args = (lp_o, undo)

# unpacking:
lp, undo = postsolve_args
c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = lp
```
Or to pack `undo` into lp_o, ie.
```Python
# definition:
postsolve_args = lp_o._replace(undo=undo)

# unpacking:
c, A_ub, b_ub, A_eq, b_eq, bounds, x0, undo = postsolve_args
```
Would you prefer one way over another?

Reply:
I prefer the former. I don't have a problem with the double nesting.

I don't think `undo` belongs in `lp_o`. (To be honest, I'm not really sure that `x0` belongs inside it, but I wouldn't change it right now.)